### PR TITLE
Fix generator crashing on declarations outside api folder

### LINKED ISSUE #1513: [Bug] `capacity overflow` on `generate` command
### Describe the bug

Hi! I'm currently following the steps to upgrade my project MomentoBooth to `flutter_rust_bridge` v2. However when running `generate` it seems like it tries to print an error but fails to do so.

```
❯ flutter_rust_bridge_codegen generate
[1.5s] Parse
  └── [0.7s] Run cargo expand
  └── [0.8s] Parse source graph
[0.0s] Generate ⠁                                                                                                             
[2023-12-28T22:11:04.834Z ERROR C:\Users\Sander\.cargo\registry\src\index.crates.io-6f17d22bba15001f\flutter_rust_bridge_codegen-2.0.0-dev.5\src\library\utils\logs.rs:52] panicked at library\alloc\src\slice.rs:503:50:                                   capacity overflow
thread 'main' panicked at library\alloc\src\slice.rs:503:50:
capacity overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The current state of the upgrade can be found [here](https://github.com/h3x4d3c1m4l/momento-booth/tree/feature/rust_bridge_v2). I pretty much just:
- Ran the `integrate` command
- Reverted any changes to code formatting on the Dart side
- Moved all API code to the `api` folder on the Rust side
- Restored `main.dart` but added the init code at the right place (at least I think so :))

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
https://pastebin.com/raw/2E0eaFvs
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Generated binding code

_No response_

### OS

Windows

### Version of `flutter_rust_bridge_codegen`

2.0.0-dev.5

### Flutter info

_No response_

### Version of `clang++`

_No response_

### Version of `ffigen`

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by h3x4d3c1m4l**
## Changes

Fixes #1513.

## Checklist

- [] An issue to be fixed by this PR is listed above.
- [ ] New tests are added to ensure new features are working. Please refer to [this page](https://cjycode.com/flutter_rust_bridge/guides/contributing/tip) to see how to add a test.
- [ ] `./frb_internal precommit --mode slow` (or `fast`) is run (it internal runs code generator, does auto formatting, etc).
- [ ] If this PR adds/changes features, documentations (in the `./website` folder) are updated.
- [ ] CI is passing. Please refer to [this page](https://cjycode.com/flutter_rust_bridge/guides/contributing/tip) to see how to solve a failed CI.

## Remark for PR creator

- `./frb_internal --help` shows utilities for development.
- If fzyzcjy does not reply for a few days, maybe he just did not see it, so please ping him.

### CODE REVIEW

File: frb_codegen/src/library/codegen/generator/api_dart/text_generator.rs [hunk 1]
```diff
@@ -47,8 +47,11 @@ fn generate_end_api_text(
         .join("\n\n");
     let classes = item.classes.iter().map(|c| c.code.clone()).join("\n\n");
 
-    // TODO use relative path calculation
-    let path_frb_generated = "../".repeat(namespace.path().len() - 2) + "frb_generated.dart";
+    let path = namespace.path();
+    let path_frb_generated = match path.len() {
+        1 => "rust/frb_generated.dart".to_string(),
```
Comment (fzyzcjy) [inline frb_codegen/src/library/codegen/generator/api_dart/text_generator.rs:52]: nit: codecov seems unhappy about this. maybe we should add a test to reproduce this scenario.

If I understand correctly, just write a function in `/Volumes/MyExternal/ExternalRefCode/flutter_rust_bridge/frb_example/pure_dart/rust/src/api/misc_example.rs`, and let that function have a struct in `src/lib.rs` as argument.

**Review by fzyzcjy — COMMENTED** [review]
Good job! Only a nit

**Comment by fzyzcjy** [comment]
Try `./frb_internal precommit --mode slow` which I guess should fix it :)

**Comment by h3x4d3c1m4l** [comment]
Mmm that command seems to end with an error on my Windows machine: https://pastebin.com/UsGge3me

**Comment by fzyzcjy** [comment]
Maybe it means that your implementation has some slight bugs? Try 

```
./frb_internal generate-run-frb-codegen-command-generate --package frb_example/pure_dart
```

which runs codegen on the example. If codegen is buggy, this command is easier to use to see what is happening.

**Comment by h3x4d3c1m4l** [comment]
I spotted a minor mistake in the `use` lines I added in the test which I corrected, but it still fails. Will take a better look tomorrow!

**Comment by fzyzcjy** [comment]
Take your time!

**Comment by fzyzcjy** [comment]
@all-contributors please add @h3x4d3c1m4l for code

**Comment by fzyzcjy** [comment]
@all-contributors please add h3x4d3c1m4l for code

**Pull request merged**